examples/ordering/run.sh

```diff
@@ -1,29 +1,88 @@
 #!/bin/bash
 
-if [ "$#" -ne 1 ]; then
-    echo "Usage: run.sh n_members"
+usage() {
+    echo "Usage: ./run.sh [-n N_NODES] [-m N_MALFUNCTIONING_NODES] [-c N_CRASHES] [-o N_ORDERED_PER_CRASH] [-d RESTART_DELAY]"
     exit 1
-fi
+}
+
+N_NODES=5
+N_MALFUNCTIONING_NODES=5
+N_CRASHES=3
+N_ORDERED_PER_CRASH=25
+RESTART_DELAY=1
+
+while getopts :n:m:c:o:d: flag; do
+    case "${flag}" in
+        n) N_NODES=${OPTARG};;
+        m) N_MALFUNCTIONING_NODES=${OPTARG};;
+        c) N_CRASHES=${OPTARG};;
+        o) N_ORDERED_PER_CRASH=${OPTARG};;
+        d) RESTART_DELAY=${OPTARG};;
+        *) usage;;
+    esac
+done
+
+n_ordered=$(( (N_CRASHES+1)*N_ORDERED_PER_CRASH ))
+port=10000
+ports="$port"
+for i in $(seq 0 $(expr $N_NODES + $N_MALFUNCTIONING_NODES - 2)); do
+    port=$((port+1))
+    ports+=",$port"
+done
 
 set -e
 
 cargo build --release
+binary="../../target/release/aleph-bft-examples-ordering"
 
 clear
 
-n_members="$1"
-ports="43000"
-port=43000
+run_crash_node () {
+    id="$1"
+    n_starting=0
+    n_data=$N_ORDERED_PER_CRASH
+    for (( i = 1; i <= N_CRASHES; i++ )); do
+        echo "Starting node $id at $n_starting items ($i/$((N_CRASHES+1)))..."
+        ! "$binary" --id "$id" --ports "$ports" --n-data "$n_data" --n-starting "$n_starting" --crash 2>> "node${id}.log"
+        echo "Node $id crashed. Respawning in $RESTART_DELAY seconds..."
+        sleep "$RESTART_DELAY"
+        n_starting=$n_data
+        n_data=$(( n_data + N_ORDERED_PER_CRASH ))
+    done
+    echo "Starting node $id at $n_starting items ($((N_CRASHES+1))/$((N_CRASHES+1)))..."
+    "$binary" --id "$id" --ports "$ports" --n-data "$n_data" --n-starting "$n_starting" 2>> "node${id}.log"
+}
 
-for i in $(seq 0 $(expr $n_members - 2)); do
-    port=$((port+1))
-    ports+=",$port"
+for id in $(seq 0 $(expr $N_NODES + $N_MALFUNCTIONING_NODES - 1)); do
+    rm -f "aleph-bft-examples-ordering-backup/${id}.units"
+    rm -f "node${id}.log"
+done
+
+
+echo "WARNING
+The current implementation of the example may halt for a number of
+reasons. In such case, try reducing the number of nodes or shortening
```

Comment:
Replace: "The current implementation of the example may halt for a number of reasons."

By "Because of a naive network network implementation, the example might fail for certain ranges of parameters (large number of nodes)."

Reply:
but it's only one of the reasons, hence "various" :P
the other one is that AlephBFT does not have censorship resistance
I'd leave it as it is, or explicitely state both reasons - what do you think?

Reply:
Ok, feel free to mention something along the lines that "AlephBFT does not strictly guarantee all input data to be included in the output -- a property that will be added in a future version"

Reply:
@DamianStraszak approved on slack, resolving